spraybot_arc_planner/src/arc_planner.cpp

```diff
@@ -0,0 +1,216 @@
+      return transformed_pose;
+    };
+
+  // Checking if the goal and start state is in the global frame
+  if (start.header.frame_id != global_frame_) {
+    RCLCPP_ERROR(
+      node_->get_logger(), "Planner will only except start position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  auto poseToYaw = [](const geometry_msgs::msg::PoseStamped p)
+    {
+      tf2::Quaternion q(
+        p.pose.orientation.x,
+        p.pose.orientation.y,
+        p.pose.orientation.z,
+        p.pose.orientation.w);
+      tf2::Matrix3x3 m(q);
+      double roll, pitch, yaw;
+      m.getRPY(roll, pitch, yaw);
+      return yaw;
+    };
+
+  geometry_msgs::msg::PoseStamped transformed_goal = transformToGlobalPose(goal);
+  geometry_msgs::msg::PoseStamped transformed_start = start;
+
+  global_path.poses.clear();
+  global_path.header.stamp = node_->now();
+  global_path.header.frame_id = global_frame_;
+
+  double distance = std::hypot(
+    transformed_goal.pose.position.x - transformed_start.pose.position.x,
+    transformed_goal.pose.position.y - transformed_start.pose.position.y);
+
+  if (distance > 2 * turn_radius_) {
+    RCLCPP_ERROR(
+      node_->get_logger(), "Computing a circular path to the goal is not possible, too small turning radius, the distance between start and goal waypoint is %f and the set radius is %f.", distance,
+      turn_radius_);
+    return global_path;
+  }
+  double theta = 2 * asin(distance / 2 / turn_radius_);
+  double alpha =
+    atan2(
+    (transformed_goal.pose.position.y - transformed_start.pose.position.y),
+    (transformed_goal.pose.position.x - transformed_start.pose.position.x));
+  double phi = ((M_PI - theta) / 2) - alpha;
+  double yaw = poseToYaw(goal);
+  double c = cos(phi);
+  double s = sin(phi);
+  int total_number_of_loop = theta / interpolation_resolution_;
+  int anti_clk {0};
+  if (((yaw >= (-M_PI)) && (yaw <= (-M_PI / 2))) or ((yaw >= M_PI / 2) && (yaw <= M_PI))) {
+    anti_clk = 1;
+  }
+
+  std::cout << "Coordinates" << std::endl;
+  std::cout << transformed_start.pose.position.x << " " << transformed_start.pose.position.y <<
+    std::endl;
+  std::cout << transformed_goal.pose.position.x << " " << transformed_goal.pose.position.y <<
+    std::endl;
+  std::cout << "Distance" << std::endl;
+  std::cout << distance << std::endl;
+  std::cout << "Theta" << std::endl;
+  std::cout << theta << std::endl;
+  std::cout << "Alpha" << std::endl;
+  std::cout << alpha << std::endl;
+  std::cout << "Phi" << std::endl;
+  std::cout << phi << std::endl;
+  std::cout << "Yaw" << std::endl;
+  std::cout << yaw << std::endl;
+  std::cout << "Direction" << std::endl;
+  std::cout << anti_clk << std::endl;
+
+  const auto & start_pos = transformed_start.pose.position;
+  const auto & goal_pos = transformed_goal.pose.position;
+
+  if (((anti_clk == 0) && (goal_pos.y > start_pos.y)) ||
+    ((anti_clk == 1) && (goal_pos.y < start_pos.y)))
```

Comment:
```suggestion
  if ((!anti_clk && goal_pos.y > start_pos.y) || (anti_clk && goal_pos.y < start_pos.y))
```
Less clustered and more readable